### PR TITLE
New Basemaps: Feature Layers Category

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

This PR is for the following sample categories:
- Feature Layers

Notes specific to this PR: 
- Feature collection layer from portal had it's indentation fixed

### General comments on Basemap updates:

It updates `Map` and/or `Basemap` Constructors as per the new design. This may include the following changes depending on the sample:
- Replace `Basemap.Type` with `BasemapStyle`
- If a lat/long/zoom was being set in the `Map` constructor, this is replaced with a `Viewpoint` being set on the `MapView`.
- Any Readme Updates identified

In addition, the samples that include the above changes, or other services requiring an API Key going forward, have the API Key  set within the sample. This will then work in conjunction with the Gradle Task being added in PR 593.

Once 593 is approved and merged into `dev` we will merge those changes into these PRs and then update all the Samples to `100.10.0-xxxx` and do a round of testing of the Samples with all the API Key / Basemap changes at that time. (We have left them as 100.9.0 (in most cases) for now to minimize merge conflicts in the `build.gradle` file).

In the meantime, the updates listed above can be reviewed. Thanks!

### CODE REVIEW

File: feature_layers/display-subtype-feature-layer/src/main/java/com/esri/samples/display_subtype_feature_layer/DisplaySubtypeFeatureLayerController.java [hunk 1]
```diff
@@ -53,10 +54,12 @@
...
 
     try {
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
 
-      // create a map with streets night vector basemap and add it to the map view
-      ArcGISMap map = new ArcGISMap();
-      map.setBasemap(Basemap.createStreetsNightVector());
+      // create a map with streets night vector basemap style and add it to the map view
```
Comment (JonLavi) [inline feature_layers/display-subtype-feature-layer/src/main/java/com/esri/samples/display_subtype_feature_layer/DisplaySubtypeFeatureLayerController.java]:
```suggestion
      // create a map with the streets night vector basemap style and add it to the map view
```

File: feature_layers/service-feature-table-manual-cache/src/main/java/com/esri/samples/service_feature_table_manual_cache/ServiceFeatureTableManualCacheSample.java [hunk 2]
```diff
@@ -117,8 +122,8 @@ public void start(Stage stage) {
         }
       });
 
-      // create a ArcGISMap with topographic basemap
-      ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      // create a map with a basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
 
       // add feature layer to the ArcGISMap
       map.getOperationalLayers().add(featureLayer);
```
Comment (JonLavi) [inline feature_layers/service-feature-table-manual-cache/src/main/java/com/esri/samples/service_feature_table_manual_cache/ServiceFeatureTableManualCacheSample.java]: feature_layers/feature-layer-query/src/main/java/com/esri/samples/feature_layer_query/FeatureLayerQuerySample.java does this in a slightly different order:
```
      // create a map with a basemap style
      final ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);

      // create a map view and set its map
      mapView = new MapView();
      mapView.setMap(map);

      // add feature layer to operational layers
      map.getOperationalLayers().add(featureLayer);
```

I quite like that one better to what's in this file, might be worth deciding between the two and sticking to one pattern?

**Review by JonLavi — APPROVED** [review]
Not much to point out here @jenmerritt , just a couple of small comments

File: feature_layers/service-feature-table-manual-cache/src/main/java/com/esri/samples/service_feature_table_manual_cache/ServiceFeatureTableManualCacheSample.java (hunk 2 above)
Comment (jenmerritt) [reply feature_layers/service-feature-table-manual-cache/src/main/java/com/esri/samples/service_feature_table_manual_cache/ServiceFeatureTableManualCacheSample.java]: Agreed, amending

File: feature_layers/feature-collection-layer/src/main/java/com/esri/samples/feature_collection_layer/FeatureCollectionLayerSample.java [hunk 3]
```diff
@@ -67,8 +68,14 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
-      // create amp and set it to be displayed in this view
-      ArcGISMap map = new ArcGISMap(Basemap.createOceans());
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with a basemap style
```
Comment (Rachael-E) [inline feature_layers/feature-collection-layer/src/main/java/com/esri/samples/feature_collection_layer/FeatureCollectionLayerSample.java]:
```suggestion
      // create a map with the oceans basemap style
```

File: feature_layers/feature-collection-layer/src/main/java/com/esri/samples/feature_collection_layer/FeatureCollectionLayerSample.java [hunk 4]
```diff
@@ -67,8 +68,14 @@ public void start(Stage stage) {
...
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_OCEANS);
+
+      // create a map view and set its map
```
Comment (Rachael-E) [inline feature_layers/feature-collection-layer/src/main/java/com/esri/samples/feature_collection_layer/FeatureCollectionLayerSample.java]:
```suggestion
      // create a map view and set the map to it
```

File: feature_layers/feature-layer-definition-expression/src/main/java/com/esri/samples/feature_layer_definition_expression/FeatureLayerDefinitionExpressionSample.java [hunk 5]
```diff
@@ -76,8 +81,8 @@ public void start(Stage stage) {
       // create feature layer from service feature table
       featureLayer = new FeatureLayer(featureTable);
 
-      // create a ArcGISMap using the basemap topographic
-      final ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+      // create a map with a basemap style
```
Comment (Rachael-E) [inline feature_layers/feature-layer-definition-expression/src/main/java/com/esri/samples/feature_layer_definition_expression/FeatureLayerDefinitionExpressionSample.java]:
```suggestion
      // create a map with the topographic basemap style
```

File: feature_layers/feature-layer-dictionary-renderer/src/main/java/com/esri/samples/feature_layer_dictionary_renderer/FeatureLayerDictionaryRendererSample.java [hunk 6]
```diff
@@ -54,7 +55,12 @@ public void start(Stage stage) {
     stage.setScene(scene);
     stage.show();
 
-    ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+    // authentication with an API key or named user is required to access basemaps and other location services
+    String yourAPIKey = System.getProperty("apiKey");
+    ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+    // create a map with a basemap style and set it to the map view
```
Comment (Rachael-E) [inline feature_layers/feature-layer-dictionary-renderer/src/main/java/com/esri/samples/feature_layer_dictionary_renderer/FeatureLayerDictionaryRendererSample.java]: there's a few more like this in this PR: I left a comment in an earlier PR to consider updating them, I'll leave it up to you if you have bandwidth to do so this time round and won't tag again in this PR. 

```suggestion
    // create a map with the topographic basemap style and set it to the map view
```

File: feature_layers/feature-layer-query/src/main/java/com/esri/samples/feature_layer_query/FeatureLayerQuerySample.java [hunk 7]
```diff
@@ -148,16 +153,16 @@ public void start(Stage stage) {
...
-      // create a ArcGISMap with basemap topographic
-      final ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
-
-      // add feature layer to operational layers
-      map.getOperationalLayers().add(featureLayer);
+      // create a map with a basemap style
+      final ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
 
-      // create a view for this ArcGISMap
+      // create a map view and set its map
```
Comment (Rachael-E) [inline feature_layers/feature-layer-query/src/main/java/com/esri/samples/feature_layer_query/FeatureLayerQuerySample.java]: a few other instances in this PR of where this comment could be slightly improved, I won't comment again to save notifications! No worries as with the basemap comment changes if you don't have bandwidth for it this time round, it's not urgent. 
```suggestion
      // create a map view and set the map to it
```

File: feature_layers/service-feature-table-manual-cache/src/main/java/com/esri/samples/service_feature_table_manual_cache/ServiceFeatureTableManualCacheSample.java [hunk 8]
```diff
@@ -117,16 +122,16 @@ public void start(Stage stage) {
...
-      map.getOperationalLayers().add(featureLayer);
+      // create a map with a basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
 
-      // create a view for this ArcGISMap and set ArcGISMap to it
+      // create a map view and set its map
       mapView = new MapView();
       mapView.setMap(map);
 
+      // add the feature layer to the operational layers
```
Comment (Rachael-E) [inline feature_layers/service-feature-table-manual-cache/src/main/java/com/esri/samples/service_feature_table_manual_cache/ServiceFeatureTableManualCacheSample.java]:
```suggestion
      // add the feature layer to the map's operational layers
```

**Review by Rachael-E — APPROVED** [review]
After looking through a few of these, I reckon we should be consistent with the basemap comment. Either: 

1. create a map with a basemap style or
2. create a map with the *insert-type* basemap style

Currently it's a mix and match. 

Also have suggested a slight improvement for the setting the map to the map view: again, it's not urgent and is super minor, so no worries if you decide there's not enough time this time round to make that change. I won't mention these two points in future PR review comments.

File: feature_layers/feature-collection-layer/src/main/java/com/esri/samples/feature_collection_layer/FeatureCollectionLayerSample.java (hunk 3 above)
Comment (jenmerritt) [reply feature_layers/feature-collection-layer/src/main/java/com/esri/samples/feature_collection_layer/FeatureCollectionLayerSample.java]: For reference, we are going with option 2: `create a map with the insert-type basemap style`

**Comment by jenmerritt** [comment]
`dev` merged in (including API Key gradle tasks changes). Samples that have an API Key set have been tested running from the IDE.

Await merging these changes until release.

**Pull request merged**